src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
The idea that dual numbers are an odd kind of complex number seems completely wrong to me. They have completely different algebraic properties: `im^2 == -1` whereas `ɛ^2 == 0` – there's no way to square that discrepancy (get it?). The fact that you can in some cases use complex numbers to do automatic differentiation of analytic functions is just an accident and doesn't mean that dual numbers are a variation on complex numbers.

In particular, the reason I think that treating dual numbers as a funny kind of real number with some extra information is that there's a field homomorphism from the dual numbers to the reals by dropping the coefficient of `ɛ`. The projection from the complex numbers to the real line by taking the real component, on the other hand, is not a homomorphism – it only preserves addition. For example:

```jl
julia> using DualNumbers

julia> real((1 + 2du)*(3 + 4du))
3

julia> real(1 + 2du)*real(3 + 4du)
3

julia> real((1 + 2im)*(3 + 4im))
-5

julia> real(1 + 2im)*real(3 + 4im)
3
```

This is because `(ℝ[ɛ]/ɛ²)/ɛ ≅ ℝ` whereas `(ℝ[i]/(i²+1))/i ≅ {0}` – i.e. you can mod out the dual numbers by ɛ and get back the real number line, whereas if you mod out the complex numbers by `i` you've collapsed 0 and -1, which forces all of the real number line to collapse to a single point.